### ISSUE TITLE
Compression filters missing from virtual datasets

### ISSUE THREAD

**Issue by axelboc** [usnistgov/h5wasm#75]
When a virtual dataset points to a compressed dataset, the `filters` information is missing from the virtual dataset's metadata. This prevents H5Web/h5wasm from knowing to load the compression plugins before attempting to read the data. See https://github.com/silx-kit/vscode-h5web/issues/43

![335981317-98589ae5-6b9b-446e-a1c0-2d3d79658502](https://github.com/usnistgov/h5wasm/assets/2936402/9638b239-c61f-4031-a2ff-1d2ff4a07106)

![image](https://github.com/usnistgov/h5wasm/assets/2936402/62260f73-e9a7-49e5-9642-fb20a76346b0)

**Comment by bmaranville** [comment]
As far as I can tell, a VDS can be built from source datasets with heterogeneous metadata (different or no chunking, different compression, dtype, etc.) - so I'm not sure that a one-to-one mirroring is possible.  We could probably add a way to get a list of all the source datasets, and then you could use that to decide on loading plugins?

**Comment by axelboc** [comment]
That seems reasonable. Do you think this could be directly included in the object returned by `get_dataset_metadata`?

```ts
interface Metadata {
  ...
  sourceDatasets?: { fileId: string; path: string }[]
}
```

**Comment by bmaranville** [comment]
Do you think there are ever virtual datasets with enough source datasets that this would become a serious performance bottleneck for reading metadata?  In h5py for instance, they have a method for retrieving source dataset metadata that is separate from reading the metadata of the virtual dataset `Dataset.virtual_sources()`

**Comment by axelboc** [comment]
Maybe at least a count to hint at whether the dataset has virtual sources? Not sure of the performance implications...

**Comment by bmaranville** [comment]
After making a preliminary implementation, it looks like it's only reading info from the dcpl of the virtual dataset, so that should be really fast (it doesn't have to resolve the source datasets), so I'm going to go with your first suggestion (but use `file_name` and `dset_name` to be more similar to what h5py puts out).

Note that sources within the same file seem to have `file_name: "."`

```js
> f.get("data_compressed_via_vds").metadata
{
  signed: false,
  type: 1,
  vlen: false,
  littleEndian: true,
  size: 8,
  total_size: 50,
  shape: [ 50 ],
  maxshape: [ 50 ],
  chunks: null,
  virtual_sources: [ { file_name: '.', dset_name: '/data_compressed' } ]
}
```

**Comment by bmaranville** [comment]
I think this is closed by eb296cb0db9e8cf066f2b2e6ff4429f286da112b (published just now as v0.7.5)
Let me know if there are any issues!

**Comment by axelboc** [comment]
Looks good to me! https://github.com/silx-kit/h5web/pull/1662

Thanks for the quick turnaround, as always 😁
